File: otp/src/main/java/com/thisjowi/otp/entity/otp.java
package com.thisjowi.otp.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class otp {

    private Long id;

    private String userId;

    private String email;

    private String secret;

    private Long expiresAt;

    private String type;

    private String issuer;

    private Integer digits;

    private Integer period;

    private String algorithm;

    private Boolean valid;

}

File: services/otp/internal/model/otp.go
package model

type Otp struct {
	ID        int64   `json:"id" db:"id"`
	UserID    string  `json:"userId" db:"user_id"`
	Email     string  `json:"email" db:"email"`
	Secret    string  `json:"secret" db:"secret"`
	ExpiresAt string  `json:"expiresAt" db:"expires_at"`
	Type      string  `json:"type" db:"type"`
	Issuer    *string `json:"issuer" db:"issuer"`
	Digits    int     `json:"digits" db:"digits"`
	Period    int     `json:"period" db:"period"`
	Algorithm *string `json:"algorithm" db:"algorithm"`
	Valid     string  `json:"valid" db:"valid"`
}

type CreateOtpRequest struct {
	Name      string `json:"name"`
	Issuer    string `json:"issuer"`
	Secret    string `json:"secret"`
	Digits    int    `json:"digits"`
	Period    int    `json:"period"`
	Algorithm string `json:"algorithm"`
	Type      string `json:"type" binding:"required"`
}
